server/schedulers/hot_region.go

```diff
@@ -779,43 +801,38 @@ func (bs *balanceSolver) calcProgressiveRank() {
 	rank := int64(0)
 	if bs.rwTy == write && bs.opTy == transferLeader {
 		// In this condition, CPU usage is the matter.
-		// Only consider about key rate.
-		srcKeyRate := srcLd.Loads[statistics.KeyDim]
-		dstKeyRate := dstLd.Loads[statistics.KeyDim]
-		peerKeyRate := peer.GetLoad(getRegionStatKind(bs.rwTy, statistics.KeyDim))
-		if srcKeyRate-peerKeyRate >= dstKeyRate+peerKeyRate {
+		// Only consider about key rate or query rate.
+		srcRate := srcLd.Loads[bs.writeLeaderFirstPriority]
+		dstRate := dstLd.Loads[bs.writeLeaderFirstPriority]
+		peerRate := peer.GetLoad(getRegionStatKind(bs.rwTy, bs.writeLeaderFirstPriority))
+		if srcRate-peerRate >= dstRate+peerRate {
 			rank = -1
 		}
 	} else {
 		firstPriorityDimHot, firstPriorityDecRatio, secondPriorityDimHot, secondPriorityDecRatio := bs.getHotDecRatioByPriorities(srcLd, dstLd, peer)
 		greatDecRatio, minorDecRatio := bs.sche.conf.GetGreatDecRatio(), bs.sche.conf.GetMinorGreatDecRatio()
 		switch {
 		case firstPriorityDimHot && firstPriorityDecRatio <= greatDecRatio && secondPriorityDimHot && secondPriorityDecRatio <= greatDecRatio:
-			// If belong to the case, both byte rate and key rate will be more balanced, the best choice.
+			// If belong to the case, two dim will be more balanced, the best choice.
 			rank = -3
-			bs.byteIsBetter = true
-			bs.keyIsBetter = true
+			bs.firstPriorityIsBetter = true
+			bs.secondPriorityIsBetter = true
 		case firstPriorityDecRatio <= minorDecRatio && secondPriorityDimHot && secondPriorityDecRatio <= greatDecRatio:
-			// If belong to the case, byte rate will be not worsened, key rate will be more balanced.
+			// If belong to the case, first priority dim will be not worsened, second priority dim will be more balanced.
 			rank = -2
-			bs.keyIsBetter = true
+			bs.secondPriorityIsBetter = true
 		case firstPriorityDimHot && firstPriorityDecRatio <= greatDecRatio:
-			// If belong to the case, byte rate will be more balanced, ignore the key rate.
+			// If belong to the case, first priority dim will be more balanced, ignore the second priority dim.
 			rank = -1
-			bs.byteIsBetter = true
+			bs.firstPriorityIsBetter = true
 		}
 	}
-	log.Debug("calcProgressiveRank",
```

Comment:
why remove debug log?

Reply:
If you use debug mode, a lot of logs will be printed and we can get this information through the firstPriorityIsBetter  metrices